nilearn/_utils/testing.py

```diff
@@ -424,44 +423,46 @@ def generate_fake_fmri(shape=(10, 11, 12), length=17, kind="noise",
          shift[1]:shift[1] + width[1],
          shift[2]:shift[2] + width[2]] = 1
 
-    if n_blocks is not None:
-        block_size = 3 if block_size is None else block_size
-        flat_fmri = fmri[mask.astype(np.bool)]
-        flat_fmri /= np.abs(flat_fmri).max()
-        target = np.zeros(length, dtype=np.int)
-        rest_max_size = (length - (n_blocks * block_size)) // n_blocks
-        if rest_max_size < 0:
-            raise ValueError(
-                '%s is too small '
-                'to put %s blocks of size %s' % (
-                    length, n_blocks, block_size))
-        t_start = 0
+    if n_blocks is None:
```

Comment:
Just cosmit refactoring: I return in this condition instead of having a huge condition block.